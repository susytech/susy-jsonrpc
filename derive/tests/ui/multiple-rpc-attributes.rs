extern crate serde;
extern crate susy_jsonrpc_core;
#[macro_use]
extern crate susy_jsonrpc_derive;

use susy_jsonrpc_core::Result;

#[rpc]
pub trait Rpc {
	/// Returns a protocol version
	#[rpc(name = "protocolVersion")]
	#[rpc(name = "protocolVersionAgain")]
	fn protocol_version(&self) -> Result<String>;
}

fn main() {}
