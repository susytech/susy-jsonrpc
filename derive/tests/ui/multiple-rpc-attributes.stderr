error: Expected only a single rpc attribute per method
  --> $DIR/multiple-rpc-attributes.rs:10:2
   |
10 |       /// Returns a protocol version
   |  _____^
11 | |     #[rpc(name = "protocolVersion")]
12 | |     #[rpc(name = "protocolVersionAgain")]
13 | |     fn protocol_version(&self) -> Result<String>;
   | |_________________________________________________^

error: aborting due to previous error

